workflow/validate/validate.go

```diff
@@ -856,11 +856,25 @@ func validateArgumentsFieldNames(prefix string, arguments wfv1.Arguments) error
 // validateArgumentsValues ensures that all arguments have parameter values or artifact locations
 func validateArgumentsValues(prefix string, arguments wfv1.Arguments, allowEmptyValues bool) error {
 	for _, param := range arguments.Parameters {
+		// check if any value is defined
 		if param.ValueFrom == nil && param.Value == nil {
 			if !allowEmptyValues {
 				return errors.Errorf(errors.CodeBadRequest, "%s%s.value is required", prefix, param.Name)
 			}
 		}
+		// check for valid valueFrom sub-parameters
+		if param.ValueFrom != nil {
+			// INFO: default needs to be accompanied by with ConfigMapKeyRef.
+			// TODO: check if ValueFrom.Expression is valid!
+			if param.ValueFrom.ConfigMapKeyRef == nil && param.ValueFrom.Expression == "" {
+				return errors.Errorf(errors.CodeBadRequest, "only default, configMapKeyRef and supplied allowed for valueFrom '%s'", param.Name)
+			}
```

Comment:
is it allowed to have `Event` and not `ConfigMapKeyRef` or `Expression`?

Reply:
I think from what I'm seeing, `Event` should be allowed but not `Expression`. 

Which IDE do you use? You should be able to do "Find all references" or the equivalent for each member of `ValueFrom` struct. That's where I can see who's calling it and how it's being used. It would be great if you might also do this to double check my findings.

Reply:
Sorry for the very late response.

You are completely right, that is the current state. Even though expression is currently not used for input/args I already had cases where I would love to use it.

Imagine you have a workflow which has "isDev" bool as input. A sub workflow now needs to set an environment dev/prod/... based on the isDev bool. Currently we need to define the sub workflow twice, set the environment parameter of each explicitly and decide on which sub workflow to trigger based on a `when: "isDev == true"` condition. In these (rare) cases it would be nice to have expression as an option for inputs. It would then be used in arguments.

Anyways, this is not the case currently.